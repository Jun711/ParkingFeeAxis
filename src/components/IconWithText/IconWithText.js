import React, { Component } from 'react'
import { View, Text } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'
import styles from './IconWithTextStyles'

export class IconWithText extends Component {

  constructor(props) {
    super(props)
    this._icon = this.props.icon || 'info'
    this._message = this.props.message || 'Loading Timeout'
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.itemContainer}>
          <View style={styles.iconContainer}>
            <Icon style={styles.infoIcon} name={this._icon}/>
          </View>
          <Text
            allowFontScaling={true}
            style={styles.infoItem}>{this._message}</Text>
        </View>
      </View>
    )
  }
}

export default IconWithText
