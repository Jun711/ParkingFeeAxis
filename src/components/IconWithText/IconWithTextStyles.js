import { StyleSheet } from 'react-native'
import { PRIMARY_FONT_SIZE, THEME_COLOR, TEXT_COLOR } from '../../util/constants'

const styles = StyleSheet.create({
  container: {
    margin: 10,
    flex: 4,
  },
  itemContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingLeft: 5,
  },
  iconContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  infoIcon: {
    fontSize: PRIMARY_FONT_SIZE,
    color: THEME_COLOR,
  },
  infoItem: {
    flex: 8,
    textAlign: 'left',
    fontSize: PRIMARY_FONT_SIZE,
    color: TEXT_COLOR,
  },
})

export default styles