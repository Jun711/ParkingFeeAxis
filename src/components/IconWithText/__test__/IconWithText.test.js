import 'react-native'
import React from 'react'
import IconWithText from '../IconWithText'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('<IconWithText />', () => {
  it('IconWithText renders correctly', () => {
    const iconWithTextComponent = renderer.create(
      <IconWithText/>
    ).toJSON()
    expect(iconWithTextComponent).toMatchSnapshot()
  })
})

