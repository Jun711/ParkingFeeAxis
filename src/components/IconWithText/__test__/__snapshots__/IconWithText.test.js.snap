// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconWithText /> IconWithText renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 4,
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingLeft": 5,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#38414e",
              "fontSize": 18,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        ï„©
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "flex": 8,
          "fontSize": 18,
          "textAlign": "left",
        }
      }
    >
      Loading Timeout
    </Text>
  </View>
</View>
`;
